### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by norogoth**
This is my submission for homework 4. Let me know what errors there are. Thank you very much for your help.

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Program.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4996407&bid=16490119)

**Comment by norogoth** [comment]
I cannot seem to find these lines in my code. I am looking at homework 4
and I only see "using system" at the top.

File: Src/BootCamp.Chapter/Program.cs [hunk 1]
```diff
@@ -1,14 +1,95 @@
...
-using Console = System.Console;
 
 namespace BootCamp.Chapter
 {
     class Program
     {
-        static void Main(string[] args)
+        public static void Main(string[] args)
         {
+            string name = PromptString("name");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: This block of code in main could be moved to a function:
PromptPersonInfo or similar.

File: Src/BootCamp.Chapter/Program.cs [hunk 2]
```diff
@@ -1,14 +1,95 @@
...
+        {
+            if (height <= 0)
+                {
+                    Console.WriteLine($"Height cannot be less than zero, but was {height}.");
+                if (weight > 0)
+                {
+                    return 0;
+                }
+                }
+            if (weight <= 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: A+ for not using else and doing validation first 🙂

**Comment by Almantask** [comment]
Not a must, but have a look at tests and try to fix the code so thst all the tests pass.

**Comment by norogoth** [comment]
> Not a must, but have a look at tests and try to fix the code so that all the tests pass.

That is a goal for me. I hope to do that once I fix all these other issues.

File: Src/BootCamp.Chapter/Program.cs (hunk 1 above)
Comment (norogoth) [reply Src/BootCamp.Chapter/Program.cs]: I am unclear on this. Isn't Prompt String the function? I am calling the function PromptString there as far as I understand.

**Comment by Almantask** [comment]
Please refer to build check and check the failing tests. I am not requiring to have 100% pass, but try a bit harder to fit the spec.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Program.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4996407&bid=16490119)

File: Src/BootCamp.Chapter/Checks.cs [hunk 3]
```diff
@@ -16,26 +16,26 @@ public static class Checks
     {
         public static int PromptInt(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            return Program.PromptInt(message);
+            //return 0;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove it completely. Commenting out code is not needed.

File: Src/BootCamp.Chapter/Checks.cs [hunk 4]
```diff
@@ -16,26 +16,26 @@ public static class Checks
...
         }
 
         public static string PromptString(string message)
         {
-            // To do: call your implementation. 
-            return "";
+            return Program.PromptString(message);
+           //return "";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove it completely. Commenting out code is not needed.

File: Src/BootCamp.Chapter/Checks.cs [hunk 5]
```diff
@@ -16,26 +16,26 @@ public static class Checks
...
         }
 
         public static float PromptFloat(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            return Program.PromptFloat(message); 
+            //return 0;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove it completely. Commenting out code is not needed.

File: Src/BootCamp.Chapter/Checks.cs [hunk 6]
```diff
@@ -16,26 +16,26 @@ public static class Checks
...
         }
 
         public static float CalculateBmi(float weight, float height)
         {
-            // To do: call your implementation. 
-            return 0;
+            return Program.CalculateBmi(weight, height);
+            //return 0;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove it completely. Commenting out code is not needed.

**Pull request closed without merging**